core/src/actor.rs

```diff
@@ -9,26 +9,79 @@ use bitcoin::{
     Address, TapSighash, TapTweakHash,
 };
 use bitcoin::{TapLeafHash, TapNodeHash, TapSighashType, TxOut};
+use bitvm::signatures::winternitz;
+
+/// Available transaction types for [`WinternitzDerivationPath`].
+#[derive(Clone, Copy, Debug)]
+pub enum TxType {
+    TimeTx,
+    KickoffTx,
+}
+
+/// Derivation path specification for Winternitz one time public key generation.
+#[derive(Debug, Clone, Copy)]
+pub struct WinternitzDerivationPath {
+    pub message_length: u32,
+    pub log_d: u32,
+    pub index: u32,
+    pub tx_type: TxType,
+    pub operator_idx: u32,
+    pub watchtower_idx: u32,
+    pub time_tx_idx: u32,
```

Comment:
Let's make all index, operator_idx, wathctower_idx and time_tx_idx Option, if it is None, then use 0 if not, the add 1 and use it. This way we can use zero indexed things